pandas/_testing/__init__.py

```diff
@@ -292,21 +296,41 @@ def makeBoolIndex(k=10, name=None):
     return Index([False, True] + [False] * (k - 2), name=name)
 
 
+def makeNumericIndex(k=10, name=None, *, dtype):
+    dtype = pandas_dtype(dtype)
+    assert isinstance(dtype, np.dtype)
+
+    if dtype.kind == "i":
+        values = list(range(k))
+    elif dtype.kind == "u":
```

Comment:
can u use np.arrange in these i and u cases 

or make the i and u cases the same structure ; they are completely different now

Reply:
I've made a better version.